### PR TITLE
purge: updated flow to allow the user to opt out

### LINKED ISSUE #21: purge playbook doesn't ask for user confirmation before starting

### LINKED ISSUE #22: purge playbook doesn't check for cephadm across the cluster hosts
Purge is reliant on cephadm binary being available on each host - but the playbook doesn't confirm that is the case before starting

### LINKED ISSUE #23: purge playbook doesn't check the validity of the fsid passed in
the fsid must exist, but we don't check

### PR DESCRIPTION
**Pull request by pcuzner**
Playbook performs some initial local checks, then checks
each ceph host has cephadm and the right fsid. Once
these plays are passed, the user is prompted to purge or
not, so they may abort

Closes #21, closes #22, closes #23

Signed-off-by: Paul Cuzner <pcuzner@redhat.com>

### CODE REVIEW

File: cephadm-purge-cluster.yml [hunk 1]
```diff
@@ -45,16 +43,79 @@
...
+      stat:
+        path: /var/lib/ceph/{{ fsid }}
+      register: fsid_exists
+      when: group_names != ['clients']
+
+    - name: fail if the fsid directory is missing
+      fail:
+        msg: >
+          The fsid directory '/var/lib/ceph/{{ fsid }}' can not be found on {{ inventory_hostname }}
+      when: fsid_exists.stat.exists == false and group_names != ['clients']
```
Comment (guits) [inline cephadm-purge-cluster.yml]:
```suggestion
      when:
        - not fsid_exists.stat.exists | bool
        - group_names != ['clients']
```

File: cephadm-purge-cluster.yml [hunk 2]
```diff
@@ -45,16 +43,79 @@
...
+      changed_when: false
+      failed_when: cephadm_exists.rc not in [ 0, 1 ]
+      when: group_names != ['clients']
+
+    - name: fail if cephadm is not available
+      fail:
+        msg: |
+          The cephadm binary is missing on {{ inventory_hostname }}. To purge the cluster you must have cephadm installed
+          on ALL ceph hosts
+      when: cephadm_exists.rc not in [ 0 ] and group_names != ['clients']
```
Comment (guits) [inline cephadm-purge-cluster.yml]:
```suggestion
      when:
        - cephadm_exists.rc not in [ 0 ]
        - group_names != ['clients']
```

**Review by guits — CHANGES_REQUESTED** [review]

File: cephadm-purge-cluster.yml (hunk 1 above)
Comment (pcuzner) [reply cephadm-purge-cluster.yml]: ACK

File: cephadm-purge-cluster.yml (hunk 2 above)
Comment (pcuzner) [reply cephadm-purge-cluster.yml]: ACK

**Comment by pcuzner** [comment]
changes applied, commits squashed

**Review by guits — APPROVED** [review]

**Comment by guits** [comment]
jenkins test all_daemons

**Comment by guits** [comment]
jenkins test deploy_and_purge

**Comment by guits** [comment]
jenkins test deploy_and_purge

**Comment by guits** [comment]
jenkins test deploy_and_purge

File: cephadm-purge-cluster.yml [hunk 3]
```diff
@@ -11,29 +11,27 @@
...
 
     - name: fail if admin group doesn't exist or is empty
       fail:
-        msg: >
+        msg: |
           You must define a group [admin] in your inventory and add a node where
           admin keyring is present at /etc/ceph/admin.client.keyring
       when: "'admin' not in groups or groups['admin'] | length < 1"
 
-- name: make sure admin keyring is present and pause cephadm
```
Comment (guits) [inline cephadm-purge-cluster.yml:32]: why did you remove this? (L32-L36)
the presence of this keyring must be checked on the node admin[0]

**Review by guits — CHANGES_REQUESTED** [review]

File: cephadm-purge-cluster.yml (hunk 3 above)
Comment (pcuzner) [reply cephadm-purge-cluster.yml:32]: a sugar low..no idea. thanks for catching. The check is in the wrong place - I had it in the localhost checks!

**Comment by guits** [comment]
@pcuzner could you squash the two first commits please?

thanks!

**Comment by pcuzner** [comment]
> @pcuzner could you squash the two first commits please?
> 
> thanks!

squashed - so you have one commit handling the playbook changes and another tox - hope that's what you meant

**Comment by guits** [comment]
> > @pcuzner could you squash the two first commits please?
> > thanks!
> 
> squashed - so you have one commit handling the playbook changes and another tox - hope that's what you meant

yup :slightly_smiling_face:

**Review by guits — APPROVED** [review]
